pkg/cmd/target.go

```diff
@@ -1020,6 +1040,49 @@ func namespaceWrapper(targetReader TargetReader, targetWriter TargetWriter, kube
 	return nil
 }
 
+//urlWrapper function target garden and shoot in dashboard url
+
+func urlWrapper(urlString string) error {
+	gardenString := ""
+	shootString := ""
+	u, err := url.Parse(urlString)
+	if err != nil {
+		fmt.Println("the URL enter is invalid")
+		os.Exit(2)
+	}
+
+	if strings.Contains(u.Host, "canary") {
```

Comment:
Don't hard-code landscape namespace here. What if my landscape is named "foo"?

Reply:
good question, is there any regular pattern between url string and garden name?

if the URL is https://dashboard.garden.foo.xxx.xxx.xxx/ then the garden name will be foo-virtual?

Reply:
no, I don't think that you can derive the garden name from the url so that it works for all cases
I would also not hardcode the landscape types like dev, staging etc. This is only valid for SAP's landscapes.

Maybe the best thing to do is to put the dashboard url in the config for each garden? This way you can uniquely identify the corresponding garden, without having any assumptions on the naming schemas. WDYT?
```yaml
gardenClusters:
  - kubeConfig: /path/kubeconfig1.yaml
    name: foo
    dashboardUrl: https://mydashboard.example.com
  - kubeConfig: /path/kubeconfig2.yaml
    name: bar
    dashboardUrl: https://bar.example.com
```

Reply:
+1 @petersutter

Reply:
i will open some internal discussion thread as for protecting sensitive information by email

Reply:
now the gardenconfig looks like this
![image](https://user-images.githubusercontent.com/42594392/79824136-0a6f8500-83c8-11ea-9256-1a56f6fa1efb.png)
 and i've modify this PR to use dashboard URL to identify garden name
thanks!